lib/internal/Magento/Framework/Lock/Backend/Database.php

```diff
@@ -0,0 +1,54 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+
+namespace Magento\Framework\lock\Backend;
+
+class Database implements \Magento\Framework\Lock\LockManagerInterface
+{
+    /** @var \Magento\Framework\App\ResourceConnection */
+    private $resource;
+
+    public function __construct(
+        \Magento\Framework\App\ResourceConnection $resource
+    )
+    {
+        $this->resource = $resource;
+    }
+
+    /**
+     * Sets a lock for name
+     *
+     * @param string $name lock name
+     * @return bool
+     */
+    public function setLock($name, $timeout = -1)
```

Comment:
As magento 2.2 and later support php 7+ only, could you scecify types for these variables? Same for other methods in this file + interface.

Also would be great to add declare strict types for new file.

Reply:
Another thing:
> MySQL 5.7.5 and later enforces a maximum length on lock names of 64 characters. Previously, no limit was enforced.

I think it should be checked in order to prevent mysql errors there

Reply:
Implemented scalar type hints.